pkg/cstorvolumereplica/v1alpha1/cstorvolumereplica.go

```diff
@@ -78,43 +80,46 @@ func (l *cvrList) GetUniquePoolNames() []string {
 	return unique
 }
 
-// listBuilder enables building
-// an instance of cvrList
-type listBuilder struct {
-	list    *cvrList
-	filters predicateList
+// ListBuilder enables building
+// an instance of CVRList
+type ListBuilder struct {
+	list    *CVRList
+	filters PredicateList
 }
 
-// ListBuilder returns a new instance
-// of listBuilder
-func ListBuilder() *listBuilder {
-	return &listBuilder{list: &cvrList{}}
+// NewListBuilder returns a new instance
+// of ListBuilder
+func NewListBuilder() *ListBuilder {
+	return &ListBuilder{list: &CVRList{}}
 }
 
 // WithAPIList builds the list of cvr
 // instances based on the provided
 // cvr api instances
-func (b *listBuilder) WithAPIList(list *apis.CStorVolumeReplicaList) *listBuilder {
+func (b *ListBuilder) WithAPIList(
+	list *apis.CStorVolumeReplicaList,
+) *ListBuilder {
 	if list == nil {
 		return b
 	}
 	for _, c := range list.Items {
-		b.list.items = append(b.list.items, cvr{object: c})
+		c := c // pin it
+		b.list.items = append(b.list.items, &CVR{object: &c})
```

Comment:
@shubham14bajpai hope you confirmed the usage of WithAPIList as you are changing from object to its reference. I mean - you need to scan entire code to check that there are no un-wanted behavior due to this change to reference.
One way - I check is - by changing the name of fn so that compiler throws error wherever code uses this fn.

Reply:
from this PR, I'm seeing this is used only in `filter` of  pkg/algorithm/cstorpoolselect/v1alpha1/select.go? is that so?

Reply:
yes I tried changing the function name and only error was in `filter` function of pkg/algorithm/cstorpoolselect/v1alpha1/select.go